src/bessel.jl

```diff
@@ -499,7 +509,7 @@ Bessel function of the first kind of order 0, ``J_0(x)``.
 """
 function besselj0(x::BigFloat)
     z = BigFloat()
-    ccall((:mpfr_j0, :libmpfr), Int32, (Ptr{BigFloat}, Ptr{BigFloat}, Int32), &z, &x, ROUNDING_MODE[])
+    ccall((:mpfr_j0, :libmpfr), Int32, (Ref{BigFloat}, Ref{BigFloat}, Int32), z, x, ROUNDING_MODE[])
```

Comment:
> z is being passed by reference, and is modified by mpfr_j0 .

Yes.


> z is copied to a Ref(z) container by ccall when it is passed to the Ref{BigFloat} argument, and the mpfr_j0 actually modifies this copy.

No.

> I think the reason is the original z is still modified is because a BigFloat object contains a pointer to the actual "limb" data, and this raw pointer was copied over to Ref(z), so z still sees the modified limbs.

So No.

> Still, it seems like it would be clearer if we initialized z = Ref(BigFloat()) and then returned z[].

The two do the same and both mutate the origional `BigFloat`. Wrapping in `Ref` makes this less clear.